fmt/container.h

```diff
@@ -0,0 +1,81 @@
+/*
+ Formatting library for C++ - standard container utilities
+
+ Copyright (c) 2012 - 2016, Victor Zverovich
+ All rights reserved.
+
+ For the license information refer to format.h.
+ */
+
+#ifndef FMT_CONTAINER_H_
+#define FMT_CONTAINER_H_
+
+#include "format.h"
+
+namespace fmt {
+
+namespace internal {
+
+/**
+  \rst
+  A "buffer" that appends data to a standard container (e.g. typically a
+  ``std::vector`` or ``std::basic_string``).
+  \endrst
+ */
+template <typename Container>
+class ContainerBuffer : public Buffer<typename Container::value_type> {
+ private:
+  Container& container_;
+  std::size_t initial_size_;
+
+ protected:
+  virtual void grow(std::size_t size) FMT_OVERRIDE {
+    container_.resize(initial_size_ + size);
+    this->ptr_ = &container_[initial_size_];
+    this->capacity_ = size;
+  }
+
+ public:
+  explicit ContainerBuffer(Container& container)
+  : container_(container), initial_size_(container.size()) {}
```

Comment:
This seems to be our Graal:

```c++
template <typename Container>
class ContainerBuffer : public Buffer<typename Container::value_type> {
 private:
  Container& container_;

 protected:
  virtual void grow(std::size_t size) FMT_OVERRIDE {
    container_.resize(size);
    this->ptr_ = &container_[0];
    this->capacity_ = size;
  }

 public:
  explicit ContainerBuffer(Container& container) : container_(container) {
    // not an "empty()" test to avoid depending on even more semantics in case
    // Container type is not standard
    if (container_.size() > 0) {
      this->ptr_ = &container_[0];
      this->size_ = container_.size();
      this->capacity_ = this->size_;
    }
  }
};
```